.github/workflows/pypi-release.yaml

```diff
@@ -0,0 +1,98 @@
+name: Build and Upload Parcels to PyPI
+on:
+  release:
+    types: [created, edited]
+  push:
+    tags:
+      - 'v*'
+  workflow_dispatch:
+
+
+jobs:
+  build-artifacts:
+    runs-on: ubuntu-latest
+    if: github.repository == 'OceanParcels/parcels'
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v3
+        name: Install Python
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install setuptools setuptools-scm wheel twine check-manifest
+
+      - name: Build tarball and wheels
+        run: |
+          git clean -xdf
+          git restore -SW .
+          python -m build --sdist --wheel .
+
+      - name: Check built artifacts
+        run: |
+          python -m twine check dist/*
+          pwd
+          if [ -f dist/parcels-0.0.0.tar.gz ]; then
+            echo "❌ INVALID VERSION NUMBER"
+            exit 1
+          else
+            echo "✅ Looks good"
+          fi
+      - uses: actions/upload-artifact@v3
+        with:
+          name: releases
+          path: dist
+
+  test-built-dist:
+    needs: build-artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v3
+        name: Install Python
+        with:
+          python-version: 3.8
+      - uses: actions/download-artifact@v3
+        with:
+          name: releases
+          path: dist
+      - name: List contents of built dist
+        run: |
+          ls -ltrh
+          ls -ltrh dist
+
+      - name: Verify the built dist/wheel is valid
+#        if: github.event_name == 'push'
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install dist/parcels*.whl
+#          python -m parcels.util.print_versions
+
+      - name: Publish package to TestPyPI
+#        if: github.event_name == 'push'
+        uses: pypa/gh-action-pypi-publish@v1.5.0
+        with:
+          user: __token__
+          password: ${{ secrets.TESTPYPI_TOKEN }}
```

Comment:
Indeed, I guess it would be easier to set this up on the parent repo, it does not seem to work on my fork

If you create an testpypi account, on https://test.pypi.org/manage/account/ you should be able to create an API token and to add it as TESTPYPI_TOKEN  in https://github.com/OceanParcels/parcels/settings/secrets/actions

Reply:
OK, just updated it